### DISCUSSION TITLE
Type 'Record<string, unknown>' is not assignable to type '{ [x: string]: {} | undefined; }'.

### DISCUSSION THREAD

**Discussion by KUBOTAfumiya** [react-hook-form/react-hook-form#10422] [Q&A]
I am trying to manage an instance of a certain package using useForm.
The instance contains the type Record<string, unknown>>, resulting in a type error.

The sample code below is a simplified version of them.
I need help on how to avoid this error.

## Sample Code
```ts
type ExampleFieldType = {
  value: Record<string, unknown>
}

const exampleField: ExampleFieldType = {
  value: {},
}

const Example = () => {
  useForm<ExampleFieldType>({
    defaultValues: exampleField,
  })
}
```

## Type Error Result
```
Type 'Record<string, unknown>' is not assignable to type '{ [x: string]: {} | undefined; }'.
  'string' index signatures are incompatible.
    Type 'unknown' is not assignable to type '{} | undefined'.
```

**Comment by joeykmh** [comment]
Anyone find a solution to this? react-hook-form's `DefaultValues` appears to cast `Record<string, unknown>` into `{ [x: string]: {} | undefined; }`

**Reply by joeykmh** [reply]
My issue is that a generated type that I do not control has changed from `Record<string, any>` to `Record<string, unknown>`. This value is passed in as a default value to some of my forms. This causes the generated type to no longer match, which breaks a ton of downstream types.

Ideally I would expect RHF to evaluate `DefaultValues<Record<string, unknown>>` to `{ [x: string]: unknown } | undefined` instead of `{} | undefined`, but maybe I'm missing some nuance about why that's bad.

My workaround now is to just manually change my manual type back to `Record<string, any>`. A more robust solution is probably to avoid passing in a type such as `Record<string, unknown>` to react-hook-form. Such a vague type is not very useful inside a library that attempts to provide strong TS support.
